var/spack/repos/builtin/packages/regcm/package.py

```diff
@@ -25,7 +25,7 @@ class Regcm(AutotoolsPackage):
     # producing a so-called fat binary. Unfortunately, gcc builds only the last
     # architecture provided (in the configure), so we allow a single arch.
     extensions = ('knl', 'skl', 'bdw', 'nhl')
-    variant('extension', default=None, values=extensions, multi=True,
+    variant('extension', default=None, values=extensions,
```

Comment:
So this doesn't actually support multiple options?

Reply:
Exactly. Non-gcc compilers (intel/pgi/..) can be instructed to build multi-arch fat binaries; as I wrote in the comment above when I submitted the package, I chose to allow a single (optional) architecture optimization.
That code `multi=True` is a leftover of when I was still experimenting, which was actually useless, since in the `configure_args` we broke out of the for loop at the first occurrence found in the spec (now I propose to go read into `self.spec.variants['extension'].value` to get the value passed, without for loops).
Allowing multiple architecture optimizations on a gcc compiler would insert `--enable-knl --enable-skl` in the configure line, and the last would be the only optimization actually produced, which would have been resulted in a wrong user experience.
It's a conservative approach, but it's the most secure.

Do you want to allow multiple arch optimizations? We could do it if we raise an error/exception if multiple archs are requested with gcc.

Reply:
I don't use this package, so I'll leave the decision up to you. I have built things with multiple architecture optimizations in the past, so if it's possible, someone might want it.

Reply:
I've talked to the developer, and I've added 2d8961a and 84274ec which allow multiple optimizations to be enabled. Unfortunately the package, as of version 4.7.x, only supports GCC and Intel: an `InstallError` is raised if a user tries to pass multiple extensions to a GCC compiler or uses a non-gcc or non-intel compilers.